pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -66,12 +77,34 @@ func New(log *logrus.Entry, restconfig *rest.Config) (Interface, error) {
 	return dh, nil
 }
 
+func (dh *dynamicHelper) resolve(groupKind, optionalVersion string) (*schema.GroupVersionResource, error) {
```

Comment:
I think it's dangerous to have two functions with effectively the same name `resolve` and `Resolve`

Reply:
Resolve() is a method of GVRResolver, while resolve() here is an local revised version of Resolve() with a call to Refresh() with resolve() fails, logically they are doing the same thing, this is why having the "same" name.

Reply:
as discussed, moved the logic back to GVRResolver.